lib/src/web_socket/browser/sockjs.dart

```diff
@@ -13,144 +13,41 @@
 import 'package:w_transport/src/web_socket/web_socket.dart';
-import 'package:w_transport/src/web_socket/web_socket_exception.dart';
 
 /// Implementation of the platform-dependent pieces of the [WebSocket] class for
 /// the SockJS browser configuration. This class uses the SockJS library to
 /// establish a WebSocket-like connection (could be a native WebSocket, could
 /// be XHR-streaming).
-class SockJSWebSocket extends CommonWebSocket implements WebSocket {
-  /// The "WebSocket" - in this case, it's a SockJS Client that has an API
-  /// similar to that of a WebSocket, regardless of what protocol is actually
-  /// used.
-  sockjs.Client _webSocket;
-
-  SockJSWebSocket._(this._webSocket, Future webSocketClosed) : super() {
-    webSocketClosed.then((closeEvent) {
-      closeCode = closeEvent.code;
-      closeReason = closeEvent.reason;
-      onIncomingDone();
-    });
-
-    // Note: We don't listen to the SockJS client for messages immediately like
-    // we do with the native WebSockets. This is because the event streams from
-    // the SockJS client are all drawn from a single broadcast stream. To make
-    // it act like a single subscription stream (and thus make it fit the
-    // interface of a standard Stream), we create a subscription when a consumer
-    // listens to this WSocket instance, cancel that subscription when the
-    // consumer's subscription is paused, and re-listen when the consumer
-    // resumes listening. See [onIncomingListen], [onIncomingPause], and
-    // [onIncomingResume].
-
-    // Additional note: the SockJS Client has no error stream, so no need to
-    // listen for errors.
-  }
-
+abstract class SockJSWebSocket extends CommonWebSocket implements WebSocket {
   static Future<WebSocket> connect(Uri uri,
       {bool debug: false,
       bool noCredentials: false,
       List<String> protocolsWhitelist,
       Duration timeout}) async {
-    Uri sockjsUri;
-    if (uri.scheme == 'ws') {
-      sockjsUri = uri.replace(scheme: 'http');
-    } else if (uri.scheme == 'wss') {
-      sockjsUri = uri.replace(scheme: 'https');
+    // The SockJS wrapper library is preferred because it uses the actual JS lib
+    // which is community-supported and fully-featured. But, it requires that
+    // the sockjs.js file is included. We can check for that by checking for the
+    // existence of a `SockJS` object on the window.
+    if (js.context['SockJS'] != null && js.context['SockJS'] is js.JsFunction) {
+      return SockJSWrapperWebSocket.connect(uri,
+          debug: debug,
+          noCredentials: noCredentials,
+          protocolsWhitelist: protocolsWhitelist,
+          timeout: timeout);
     }
 
-    final client = new sockjs.Client(sockjsUri.toString(),
-        debug: debug == true,
-        noCredentials: noCredentials == true,
+    // If the sockjs.js file wasn't detected, then we fallback to the original
+    // SockJS Dart port.
+    return SockJSPortWebSocket.connect(uri,
```

Comment:
I added a note to the changelog, the source docs for `WebSocket`, and the SockJS guide. I'd rather not log/print anything right now since I don't see us making the major jump anytime soon and in the meantime it's valid for consumers to not want to change anything since it still works.